### ISSUE TITLE
Call to Cake\ORM\Table::belongsTo could not find the class for "x"

### ISSUE THREAD

**Issue by mixable** [CakeDC/cakephp-phpstan#30]
I have a table association defined as follows:
```php
$this->belongsTo('Thumbs')
    ->setClassName('Medias')
    ->setForeignKey('media_id');
```

Since version **[3.1.0](https://github.com/CakeDC/cakephp-phpstan/releases)**, I get the following error when running phpstan:
```
 ------ ----------------------------------------------------------------------------------------- 
  Line   src/Model/Table/AdditivesTable.php                                     
 ------ ----------------------------------------------------------------------------------------- 
  54     Call to App\Model\Table\AdditivesTable::belongsTo could not find the class for "Thumbs"  
 ------ ----------------------------------------------------------------------------------------- 
```

The same error is also raised for other associations that use `belongsToMany`, `hasMany` and `belongsTo` when a different class is used with `setClassName()`.

I didn't check the code in detail, but I think `AddAssociationExistsTableClassRule` does not handle the value of `setClassName()` correctly. With this, the associated class might be different from the provided associated name.

I expected not to get an error for this association, because the class provided by `setClassName()` exists.

**Comment by rochamarcelo** [comment]
This is being handled.

**Comment by dereuromark** [comment]
https://github.com/CakeDC/cakephp-phpstan/pull/31

**Comment by rochamarcelo** [comment]
@mixable Issue is fixed, please update to 3.1.2

**Comment by mixable** [comment]
Thank you! Yes, I can confirm that running phpstan works as expected now.
